lib/Caxy/HtmlDiff/HtmlDiff.php

```diff
@@ -209,7 +222,10 @@ protected function processEqualOperation($operation)
                 if (in_array($s, $this->isolatedDiffTags) && isset($this->newIsolatedDiffTags[$pos])) {
                     $oldText = implode("", $this->findIsolatedDiffTagsInOld($operation, $pos));
                     $newText = implode("", $this->newIsolatedDiffTags[$pos]);
-                    $result[] = $this->diffElements($oldText, $newText);
+                    $type = in_array($s, array($this->isolatedDiffTags['ol'], $this->isolatedDiffTags['dl'], $this->isolatedDiffTags['ul']))
+                        ? "List"
+                        : "Elements";
+                    $result[] = $this->{'diff' . $type}($oldText, $newText);
```

Comment:
I'm not sure that I'm sold on calling functions dynamically like this yet -- mainly because it breaks things like Scrutinizer and IDE auto-completion. I definitely understand its value, but I think in this situation since there are only two cases it may be worth using an if / else and calling the function explicitly so that we get the Scrutinizer checking and auto-completion.

This is just what I'm thinking, not basing it on any concrete best practices that I know of -- so if you disagree I'm all ears :smiley: